docs/JestObjectAPI.md

```diff
@@ -568,6 +568,32 @@ jest.isolateModules(() => {
 const otherCopyOfMyModule = require('myModule');
 ```
 
+### `jest.isolateModulesAsync(fn)`
+
+`jest.isolateModulesAsync()` is the equivalent of `jest.isolateModules()`, but for async functions that need to be wrapped. The caller is expected to `await` the completion of `isolateModulesAsync`.
+
+```js
+let myModule;
+await jest.isolateModulesAsync(async () => {
+  myModule = await import('myModule');
+  // do async stuff here
+});
+
+const otherCopyOfMyModule = require('myModule');
+```
+
+`jest.isolateModulesAsync()` can also run synchronous callbacks:
```

Comment:
Fine with that, but why to have two identical APIs? How the reader should decide which one to use? Also `await` looks like a mistake if a method is not thenable. See: https://typescript-eslint.io/rules/await-thenable/

Isn’t it more clear to say: use `jest.isolateModules(fn)` for sync operation and go for `jest.isolateModulesAsync()` if you want to deal with async stuff.

Reply:
Fine with me, I'll revert the sync changes.